app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,14 @@
+class DishesController < ApplicationController
+  def show
+    @dish = Dish.find(params[:id])
+  end
+
+  # I would like to know how I could do the update action 'better' feels very rudimentary 
+  def update
+    @dish = Dish.find(params[:id])
+    @ingredient = Ingredient.find(params[:add_ingredient])
+    @dish.ingredients << @ingredient
+
+    redirect_to "/dishes/#{@dish.id}"
+  end
```

Comment:
Let me start by just mentioning that since the `update` action doesn't send anything to a view, you don't need to set everything to an instance variable.

The only other thing I would call out is that this method, `update`, should be `create` and it should live on either the child controller `IngredientsController` or a join controller `DishIngredientsController`. If we think about what it's doing in the background it is creating a relationship between a child and the parent, it's not updating the parent.

The code itself is perfect and I don't think there is anything "better" I'd suggest.

You could do something like this if you wanted:

```ruby
dish = Dish.find(params[:id])
DishIngredient.create(dish: dish, ingredient: params[:add_ingredient])

redirect_to dish_path(dish)
```
Saves one line, I guess 🤷